hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java

```diff
@@ -34,18 +39,39 @@ public class ConsensusCreateTopicHandler implements TransactionHandler {
      * required keys, warms the cache, and creates the {@link TransactionMetadata} that is used in
      * the handle stage.
      *
-     * <p>Please note: the method signature is just a placeholder which is most likely going to
-     * change.
-     *
      * @param txBody the {@link TransactionBody} with the transaction data
      * @param payer the {@link AccountID} of the payer
+     * @param keyLookup the {@link AccountKeyLookup} to use for key lookups
      * @return the {@link TransactionMetadata} with all information that needs to be passed to
      *     {@link #handle(TransactionMetadata)}
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txBody,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup) {
+        final var metaBuilder =
+                new SigTransactionMetadataBuilder(keyLookup).txnBody(txBody).payerKeyFor(payer);
+
+        final var op = txBody.getConsensusCreateTopic();
+        final var adminKey = asHederaKey(op.getAdminKey());
+        adminKey.ifPresent(metaBuilder::addToReqNonPayerKeys);
+        final var submitKey = asHederaKey(op.getSubmitKey());
+        submitKey.ifPresent(metaBuilder::addToReqNonPayerKeys);
+
+        if (op.hasAutoRenewAccount()) {
+            final var autoRenewAccount = op.getAutoRenewAccount();
+            if (!payer.equals(autoRenewAccount)) {
+                final var result = keyLookup.getKey(autoRenewAccount);
+                if (!result.failed() && result.key() != null) {
+                    metaBuilder.addToReqNonPayerKeys(result.key());
+                } else {
+                    metaBuilder.status(ResponseCodeEnum.INVALID_AUTORENEW_ACCOUNT);
```

Comment:
The payer not equals check is already done in `addNonPayerKey` method. So you can do this one line instead of all these lines
```
metaBuilder.addNonPayerKey(autoRenewAccount, INVALID_AUTORENEW_ACCOUNT);
```